code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/AEPPushTemplate.kt

```diff
@@ -98,6 +143,10 @@ internal sealed class AEPPushTemplate {
     internal var actionButtonsString: String?
         private set
 
+    // Optional, list of ActionButton for the notification
+    internal var actionButtonsList: List<ActionButton>?
+        private set
+
```

Comment:
Given that this is the base template for all other templates, it should only contain attributes that are common to _all_ templates. I believe not all templates allow action buttons. So I prefer not to have the action button related logic here in this model class.

Reply:
Action buttons are only supported in basic template type. Moving this to `BasicPushTemplate`